internal/dao/sqlx/reservation_dao.go

```diff
@@ -174,11 +212,11 @@ func (di *reservationDaoSqlx) ListInstances(ctx context.Context, limit, offset i
 	return result, nil
 }
 
-func (di *reservationDaoSqlx) UpdateStatus(ctx context.Context, id int64, status string) error {
+func (di *reservationDaoSqlx) UpdateStatus(ctx context.Context, id int64, status string, addSteps int32) error {
```

Comment:
Do we expect adding more steps at the same time? or is that for not adding a step, but still update status?

Reply:
This function is meant to update status string and step number in a relative way (typically 0 or 1 - add nothing, add one). If you have a better idea, let me know. I do not love this design honestly, but I could not figure out anything better since you want to update status text (beginning of a job) and status and step (end of a job). Two functions look like an overkill (two queries).

Reply:
I did not realize we might not want to bumpt step when updating status, this is good :+1: two functions would be indead overkill for one argument difference, I just missed the second use case :+1: